ember-string/src/index.ts

```diff
@@ -52,20 +53,20 @@ const CLASSIFY_CACHE = new Cache<string, string>(1000, (str) => {
     _match: string,
     initialChar: string,
     _separator: string,
-    chr: string
+    chr: string,
   ) => initialChar + (chr ? chr.toUpperCase() : '');
   const parts = str.split('/');
 
   for (let i = 0; i < parts.length; i++) {
-    parts[i] = parts[i]
+    parts[i] = (parts as any)[i]
```

Comment:
added because TS has gotten stricter since this PR was initially opened and TS now considers this dangerous -- it would be great though if it could determine that `i` was bounded to the range within `parts`